developer-docs-site/static/messageboard_demo/client/index.ts

```diff
@@ -0,0 +1,172 @@
+// 1. compiling the modules with the admin account's address
+// 2. publishing the modules to the devnet
+// 3. sending txn to the messageboard
+// 4. reading the events emitted from the messageboard
+// Note: you need to install the typescript SDK in the ecosystem/typescript folder first.
+
+import {AptosClient, AptosAccount, FaucetClient, Types, HexString} from "aptos";
+import * as fs from 'fs';
+import * as path from 'path';
+import {Buffer} from 'buffer';
+import {exec} from "child_process";
+import {chdir} from 'process';
+
+
+const NODE_URL = "https://fullnode.devnet.aptoslabs.com";
+const FAUCET_URL = "https://faucet.devnet.aptoslabs.com";
+
+enum BoardType {
+    ACL = "ACLBasedMB"
+}
+
+class Messageboard {
+    boardType: BoardType;
+    client: AptosClient;
+    role: string;
+    adminAddr: HexString;
+    admin: AptosAccount;
+
+    constructor(boardType: BoardType, admin: AptosAccount) {
+        this.boardType = boardType;
+        this.client = new AptosClient(NODE_URL);
+        this.admin = admin;
+        this.adminAddr = admin.address();
+    }
+
+    async executeTransaction(
+        account: AptosAccount,
+        payload: Types.TransactionPayload
+    ): Promise<Types.HexEncodedBytes> {
+        var txnRequest = await this.client.generateTransaction(account.address(), payload);
+        var signedTxn = await this.client.signTransaction(account, txnRequest);
+        var transactionRes = await this.client.submitTransaction(account, signedTxn);
+        await this.client.waitForTransaction(transactionRes.hash);
+        return transactionRes.hash;
+    }
+
+    getCompiledModuleTxnPayload(dirPath: string, names: Set<string>): Types.TransactionPayload {
+        let files = [];
+
+        function visit_dir(d: string) {
+            fs.readdirSync(d).forEach(f => {
+                const a = path.join(d, f);
+                if (fs.statSync(a).isDirectory()) return visit_dir(a);
+                else {
+                    if (names.has(path.basename(a))) {
+                        files.push(a);
+                    }
+                    return;
+                }
+            });
+        }
+
+        visit_dir(dirPath);
+        console.log(files);
+        let modules = files.map(function (e) {
+            const moduleHex = fs.readFileSync(e).toString("hex");
+            return {"bytecode": `0x${moduleHex}`}
+        });
+
+        const payload: Types.TransactionPayload = {
+            type: "module_bundle_payload",
+            modules: modules,
+        };
+        return payload;
+    }
+
+    getScriptFunctionTxnPayload(funcName: string, args: Types.MoveValue[]): Types.TransactionPayload {
+        const payload = {
+            type: "script_function_payload",
+            function: `${funcName}`,
+            type_arguments: [],
+            arguments: args
+        };
+        return payload;
+    }
+
+    async createMessageboard(account: AptosAccount, modulePath: string) {
```

Comment:
this should be two different steps....

Ideally we should have "installMessageboardModule" and "createMessageboard" separately.

installation happens only once, but there's no reason why we would ever need more than one instance of the module installed. Whereas Alice and Bob can use the same module but call "createMessageboard" using the same backend. Much like we did in the first move module tutorial.

Reply:
split to two